mfr/providers/osf/provider.py

```diff
@@ -66,23 +66,24 @@ def __init__(self, request, url):
             metadata_url = download_url.replace('/file?', '/data?', 1)
             metadata_request = await self._make_request('GET', metadata_url)
             metadata = await metadata_request.json()
+            await metadata_request.release()
         else:
             # URL is for WaterButler v1 API
             self.metrics.add('metadata.wb_api', 'v1')
             metadata_request = await self._make_request('HEAD', download_url)
-            # To make changes to current code as minimal as possible
-            try:
-                metadata = {'data': json.loads(metadata_request.headers['x-waterbutler-metadata'])['attributes']}
-                await metadata_request.release()
-            except KeyError:
-                raise exceptions.MetadataError(
-                    'Failed to fetch metadata. Received response '
-                    'code {}'.format(str(metadata_request.status)),
+            request_code = metadata_request.status
+            request_reason = metadata_request.reason
+            request_headers = metadata_request.headers
+            await metadata_request.release()
+            if request_code != 200:
+                raise exceptions.MetadataError('Failed to fetch file metadata from WaterButler. Received response: code {} {}'.format(str(request_code), str(request_reason)),
                     metadata_url=download_url,
-                    response=await metadata_request.text(),
+                    response=request_reason,
```

Comment:
Does `response=` expect the response body instead of the status message? Status message is probably good enough but I just want to confirm this change. Here is the reference: `.reason` ([link](http://aiohttp.readthedocs.io/en/stable/client_reference.html#aiohttp.ClientResponse.reason)) and `.text()` ([link](http://aiohttp.readthedocs.io/en/stable/client_reference.html#aiohttp.ClientResponse.text)).

Reply:
It's expecting text. I'm afraid I can't recall why I changed it, and I don't see where the exception actually does anything with it.